staging/src/k8s.io/component-helpers/storage/volume/pv_helpers.go

```diff
@@ -227,6 +229,11 @@ func FindMatchingVolume(
 			continue
 		}
 
+		// filter out mismatching volumeAttributesClassName
```

Comment:
Some thing here, I think we should check for VAC name match regardless of feature gate enablement.